File: 프로그래머스/1/12917. 문자열 내림차순으로 배치하기/문자열 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string s) {
    string answer = "";
    
    string largeAlpha = "";
    string smallAlpha = "";
    for(int i=0; i<s.length(); ++i)
    {
        if(s[i] >= 65 && s[i] <= 90)
            largeAlpha += s[i];
        else
            smallAlpha += s[i];
    }
    sort(largeAlpha.rbegin(), largeAlpha.rend());
    sort(smallAlpha.rbegin(), smallAlpha.rend());
    answer += smallAlpha;
    answer += largeAlpha;
    
    return answer;
}

File: 프로그래머스/1/12917. 문자열 내림차순으로 배치하기/문자열 내림차순으로 배치하기.cs
using System;

public class Solution {
    public string solution(string s) {
        string answer = "";
        
        int strLength = s.Length;
        char[] cArr = new char[strLength];
        for(int i = 0; i < strLength; ++i)
            cArr[i] = s[i];
        Array.Sort(cArr);
        Array.Reverse(cArr);
        
        for(int i = 0; i < s.Length; ++i)
            answer += cArr[i];
        
        return answer;
    }
}